prometheus-formula/prometheus/files/prometheus-rules.yml

```diff
@@ -0,0 +1,158 @@
+    for: 10m
+    labels:
+      severity: warning
+  - alert: PrometheusTSDBReloadsFailing
+    annotations:
+      description: Prometheus {{$labels.instance}} has detected
+        {{$value | humanize}} reload failures over the last 3h.
+      summary: Prometheus has issues reloading blocks from disk.
+    expr: |
+      increase(prometheus_tsdb_reloads_failures_total{job="prometheus"}[3h]) > 0
+    for: 4h
+    labels:
+      severity: warning
+  - alert: PrometheusTSDBCompactionsFailing
+    annotations:
+      description: Prometheus {{$labels.instance}} has detected
+        {{$value | humanize}} compaction failures over the last 3h.
+      summary: Prometheus has issues compacting blocks.
+    expr: |
+      increase(prometheus_tsdb_compactions_failed_total{job="prometheus"}[3h]) > 0
+    for: 4h
+    labels:
+      severity: warning
+  - alert: PrometheusNotIngestingSamples
+    annotations:
+      description: Prometheus {{$labels.instance}} is not ingesting samples.
+      summary: Prometheus is not ingesting samples.
+    expr: |
+      rate(prometheus_tsdb_head_samples_appended_total{job="prometheus"}[5m]) <= 0
+    for: 10m
+    labels:
+      severity: warning
+  - alert: PrometheusDuplicateTimestamps
+    annotations:
+      description: Prometheus {{$labels.instance}} is dropping
+        {{ printf "%.4g" $value  }} samples/s with different values but duplicated
+        timestamp.
+      summary: Prometheus is dropping samples with duplicate timestamps.
+    expr: |
+      rate(prometheus_target_scrapes_sample_duplicate_timestamp_total{job="prometheus"}[5m]) > 0
+    for: 10m
+    labels:
+      severity: warning
+  - alert: PrometheusOutOfOrderTimestamps
+    annotations:
+      description: Prometheus {{$labels.instance}} is dropping
+        {{ printf "%.4g" $value  }} samples/s with timestamps arriving out of order.
+      summary: Prometheus drops samples with out-of-order timestamps.
+    expr: |
+      rate(prometheus_target_scrapes_sample_out_of_order_total{job="prometheus"}[5m]) > 0
+    for: 10m
+    labels:
+      severity: warning
+  - alert: PrometheusRuleFailures
+    annotations:
+      description: Prometheus {{$labels.instance}} has failed to evaluate 
+        {{ printf "%.0f" $value }} rules in the last 5m.
+      summary: Prometheus is failing rule evaluations.
+    expr: |
+      increase(prometheus_rule_evaluation_failures_total{job="prometheus"}[5m]) > 0
+    for: 15m
+    labels:
+      severity: critical
+  - alert: PrometheusMissingRuleEvaluations
+    annotations:
+      description: Prometheus {{$labels.instance}} has missed {{
+        printf "%.0f" $value }} rule group evaluations in the last 5m.
+      summary: Prometheus is missing rule evaluations due to slow rule group evaluation.
+    expr: |
+      increase(prometheus_rule_group_iterations_missed_total{job="prometheus"}[5m]) > 0
+    for: 15m
+    labels:
+      severity: warning
+- name: general.rules
```

Comment:
Do we want to include target down in meta rules? This might generate duplicate alerts as most users will set their own target down rules

Reply:
Agree that it doesn't really fit into meta-monitoring rules file, but we should include this rule in some form. How about moving it to a separate rules file _general_rules.yml_? Later we could add more default rules for the exporters we ship as a part of another story.

Reply:
That sounds good.